lib/twiglet/message.rb

```diff
@@ -0,0 +1,29 @@
+module Twiglet
+  class Message
+    attr_reader :content
+
+    def self.format(msg)
+      new(msg).content
+    end
+
+    def initialize(msg)
+      case msg
+      when String
+        @content = { message: msg }
+      when Hash
+        @content = msg.transform_keys!(&:to_sym)
+      else
+        raise('Message must be String or Hash')
+      end
+
+      validate!
+    end
+
+    private
+
+    def validate!
```

Comment:
Would these guards not be better placed inline in the initialiser?

Reply:
The initializer was getting a little long/had a lot of conditional logic IMO, but I moved the other `raise` into the `validate!` method to better separate responsibilities.

Reply:
OK, could we make the three lines of raises in the validate method the same then? i.e. all `raise x if thing?`